source/common/network/address_impl.cc

```diff
@@ -404,10 +404,11 @@ absl::Status PipeInstance::initHelper(const sockaddr_un* address, mode_t mode) {
 }
 
 EnvoyInternalInstance::EnvoyInternalInstance(const std::string& address_id,
+                                             const std::string& endpoint_id,
                                              const SocketInterface* sock_interface)
     : InstanceBase(Type::EnvoyInternal, sockInterfaceOrDefault(sock_interface)),
-      internal_address_(address_id) {
-  friendly_name_ = absl::StrCat("envoy://", address_id);
+      internal_address_(address_id, endpoint_id) {
+  friendly_name_ = absl::StrCat("envoy://", address_id, ":", endpoint_id);
```

Comment:
Given the endpoint_id in the real world contains ":" (ipv6 address and port). It might be good to use another separator here.

Reply:
Sure, will use slash instead.